kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/ConfigurationRegistar.java

```diff
@@ -0,0 +1,54 @@
+/**
+ *  Copyright 2005-2016 Red Hat, Inc.
+ *
+ *  Red Hat licenses this file to you under the Apache License, version
+ *  2.0 (the "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ *  implied.  See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+package org.arquillian.cube.kubernetes.impl;
+
+import org.arquillian.cube.impl.util.Strings;
+import org.arquillian.cube.kubernetes.api.Configuration;
+import org.jboss.arquillian.config.descriptor.api.ArquillianDescriptor;
+import org.jboss.arquillian.core.api.InstanceProducer;
+import org.jboss.arquillian.core.api.annotation.ApplicationScoped;
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.core.api.annotation.Observes;
+
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.arquillian.cube.kubernetes.impl.Constants.DEFAULT_MAVEN_PROTOCOL_HANDLER;
+import static org.arquillian.cube.kubernetes.impl.Constants.JAVA_PROTOCOL_HANDLER;
+import static org.arquillian.cube.kubernetes.impl.Constants.PROTOCOL_HANDLERS;
+
+public class ConfigurationRegistar {
+
+    private static final String EXTENSION_NAME = "kubernetes";
+
+    @Inject
+    @ApplicationScoped
+    private InstanceProducer<Configuration> configurationProducer;
+
+    public void configure(@Observes ArquillianDescriptor arquillianDescriptor) {
+        Map<String, String> config = arquillianDescriptor.extension(EXTENSION_NAME).getExtensionProperties();
+        configurationProducer.set(DefaultConfiguration.fromMap(config));
+        configureProtocolHandlers(config);
+    }
+    
+    private static void configureProtocolHandlers(Map<String, String> conf) {
+        Set<String> handlers = new LinkedHashSet<>();
+        handlers.addAll(Strings.splitAndTrimAsList(System.getProperty(JAVA_PROTOCOL_HANDLER, ""), " "));
+        handlers.addAll(Strings.splitAndTrimAsList(conf.containsKey(PROTOCOL_HANDLERS) ? conf.get(PROTOCOL_HANDLERS) : DEFAULT_MAVEN_PROTOCOL_HANDLER, " "));
+        System.setProperty(JAVA_PROTOCOL_HANDLER, Strings.join(handlers, " "));
```

Comment:
I see that this property is set but never clear. Is this normal or safe? I am thinking in case of splitting tests in parallel in same VM for example.